### PR TITLE
fix: Merkle tree bug fix

### PR DESCRIPTION
**Pull request by mrain**
## Description

closes: #499 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (main)
- [x] Linked to GitHub issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [x] Wrote unit tests
- [ ] Updated relevant documentation in the code
- [ ] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer

### CODE REVIEW

**Review by jbearer — DISMISSED** [review]

**Comment by mrain** [comment]
Reason for previous CI failure is [here](https://github.com/yaahc/displaydoc/issues/46)

A temporarily fix is to use stable cargo for test [d6439a3](https://github.com/EspressoSystems/jellyfish/pull/500/commits/d6439a3415974ef11b66ca49ee1f1644f2a3bf1b)

**Review by jbearer — APPROVED** [review]

File: primitives/src/merkle_tree/internal.rs [hunk 1]
```diff
@@ -278,7 +288,19 @@ where
...
+        Ok((
+            Arc::new(MerkleNode::Leaf {
+                value: H::digest_leaf(&0, leaves[0].borrow())?,
+                pos: 0,
+                elem: leaves[0].borrow().clone(),
+            }),
+            height,
+            1,
+        ))
+    } else {
```
Comment (ggutoski) [inline primitives/src/merkle_tree/internal.rs:303]: copy pasta 😕

File: primitives/src/merkle_tree/light_weight.rs [hunk 2]
```diff
@@ -174,6 +174,11 @@ mod mt_tests {
     }
 
     fn test_light_mt_lookup_helper<F: RescueParameter>() {
+        // singleton merkle tree test (#499)
+        let mt = RescueLightWeightMerkleTree::<F>::from_elems(None, [F::from(0u64)]).unwrap();
+        let (elem, _) = mt.lookup(0).expect_ok().unwrap();
+        assert_eq!(elem, &F::from(0u64));
```
Comment (ggutoski) [inline primitives/src/merkle_tree/light_weight.rs:180]: more copy pasta

File: scripts/run_tests.sh [hunk 3]
```diff
@@ -4,7 +4,7 @@ set -e
 # We want the code to panic if there is an integer overflow
 export RUSTFLAGS="-C overflow-checks=on"
 
-cargo +nightly test --release -p jf-utils -- -Zunstable-options --report-time
-cargo +nightly test --release -p jf-plonk --lib --bins -- -Zunstable-options --report-time
-cargo +nightly test --release -p jf-primitives --features test-srs -- -Zunstable-options --report-time # enable test-srs feature for gen_srs_for_testing
-cargo +nightly test --release -p jf-relation -- -Zunstable-options --report-time
```
Comment (ggutoski) [inline scripts/run_tests.sh:10]: thanks for removing nightly!

**Review by ggutoski — APPROVED** [review]

**Pull request merged**